lib/sparse/src/common/scores_memory_pool.rs

```diff
@@ -0,0 +1,66 @@
+use common::types::ScoreType;
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+
+type PooledScores = Vec<ScoreType>;
+
+lazy_static! {
+    /// Max number of `PooledScores` to preserve in memory
+    ///
+    /// If there are more concurrent requests, a new temporary vec is created dynamically.
+    /// This limit is implemented to prevent memory leakage.
+    /// It matches the number of logical CPU cores, to best represent the expected number of
+    /// concurrent requests. Clamped between 16 and 128 to prevent extreme values.
+    static ref POOL_KEEP_LIMIT: usize = common::cpu::get_num_cpus().clamp(16, 128);
+}
```

Comment:
I like the idea of moving it to `default.rs` :+1:

Reply:
This is the same as in visited pool in HNSW. May we use already defined limit with the same value instead of new one?

Reply:
> This is the same as in visited pool in HNSW. May we use already defined limit with the same value instead of new one?

Yes. That's basically what I suggested. :smiley:

Reply:
Yep, true😃 didn't read your suggestion properly =)